ci/kubetest/test_clickhouse_sharding.py

```diff
@@ -48,17 +50,15 @@ def test_upgrading_to_more_shards(kube):
     new_values["clickhouse"]["layout"]["shardsCount"] = 3
 
     install_chart(new_values)
-    wait_for_pods_to_be_ready(
-        kube,
-        labels={
-            "clickhouse.altinity.com/namespace": "posthog",
-            "clickhouse.altinity.com/chi": "posthog",
-        },
-        expected_count=6,
-    )
-
-    # Validate the ClickHouse node setup
-    number_of_hosts, table_counts = get_clickhouse_table_counts_on_all_nodes(kube)
+
+    # Wait for new replicas to come up and for tables to be created
+    for attempt in range(24):
```

Comment:
Nit: I usually prefer to set a deadline so that the overall timeframe is fixed

```
start = time.time()
while time.time() - start < something: # e.g. 10 minutes
```